### PR TITLE
[ios] Fixed bookmark list descriptions

### LINKED ISSUE #4799: [iOS] List description isn't showing anywhere in the app
**Describe the issue**
The list description isn't showing anywhere in the app.

---

**Expected behaviour**
I assume the description should be shown under the list title where is the blue text `DESCRIPTION` located.

---

https://user-images.githubusercontent.com/66956532/225744810-7b9a28cb-48ff-4918-8dfc-f1bece9310f9.mp4

---

**System information:**
 - Operating system and its version: iOS 16.3.1
 - Organic Maps version: `2023.03.02-1`
 - Device Model: iPhone SE 2020

---

**Additional context**
After unzipping the KMZ file, we can see the description data is saved there, so it's just a visualisation problem, I assume:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<kml xmlns="http://earth.google.com/kml/2.2">
<Document>
…
  <name>Test</name>
  <description>List description</description>
…
```

### PR DESCRIPTION
**Pull request by andre1110**
Fixes #4799

### CODE REVIEW

**Review by biodranik — COMMENTED** [review]
Thanks! Can you please record a video of how it looks and works now for text and HTML?

**Comment by andre1110** [comment]
See attached. Please let me know what you think.
Simulator Screen Recording - iPhone 14 Pro - 2023-06-10 at 15 39 13

File: iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm [hunk 1]
```diff
@@ -287,6 +287,11 @@ - (BOOL)hasExtraInfo:(MWMMarkGroupID)groupId {
   return !data.m_description.empty() || !data.m_annotation.empty();
 }
 
+- (BOOL)isHtmlDescription:(MWMMarkGroupID)groupId {
+    auto description = GetPreferredBookmarkStr(self.bm.GetCategoryData(groupId).m_description);
```
Comment (biodranik) [inline iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm:291]:
```suggestion
    auto const description = GetPreferredBookmarkStr(self.bm.GetCategoryData(groupId).m_description);
```

File: iphone/Maps/Bookmarks/BookmarksList/BookmarksListInfoViewController.swift [hunk 2]
```diff
@@ -26,17 +26,23 @@ final class BookmarksListInfoViewController: UIViewController {
...
     guard let info = info else { return }
     updateInfo(info)
   }
 
-  private func updateInfo(_ info: IBookmakrsListInfoViewModel) {
+  private func updateInfo(_ info: IBookmarksListInfoViewModel) {
     titleLabel.text = info.title
     descriptionButton.isHidden = !info.hasDescription
-
+    descriptionButton.isEnabled = info.isHtmlDescription
```
Comment (biodranik) [inline iphone/Maps/Bookmarks/BookmarksList/BookmarksListInfoViewController.swift:38]: What if a non-html description is big enough? Will it expand in the disabled state?

File: iphone/Maps/Bookmarks/BookmarksList/BookmarksListInfoViewController.swift [hunk 3]
```diff
@@ -26,17 +26,23 @@ final class BookmarksListInfoViewController: UIViewController {
...
+    if info.hasDescription {
+      var description = info.isHtmlDescription
+      ? BookmarksListInfoViewController.getPlainText(info.description)
+      : info.description
+      descriptionButton.setTitle(description?.uppercased(), for: .normal)
```
Comment (biodranik) [inline iphone/Maps/Bookmarks/BookmarksList/BookmarksListInfoViewController.swift:43]:
```suggestion
      let description = info.isHtmlDescription
          ? BookmarksListInfoViewController.getPlainText(info.description)
          : info.description
      descriptionButton.setTitle(description for: .normal)
```
There is no need to uppercase it.

File: iphone/Maps/Bookmarks/BookmarksList/BookmarksListInfoViewController.swift (hunk 2 above)
Comment (andre1110) [reply iphone/Maps/Bookmarks/BookmarksList/BookmarksListInfoViewController.swift:38]: It won't. I think that we should distinguish 2 use cases here:

1. A user specified an html string - in this case we should certainly allow to open a new WebView to display rich formatting, pictures, etc.
2. A user set a plain string, which I assume to be really short in majority of cases. Thus, it'd be excessive to open a WebView for this

As an option, we can 1) enable the button only if the string size will be more than it's possible to fit the string using [NSString.size(withAttributes:)](https://developer.apple.com/documentation/foundation/nsstring/1531844-size), or
2) shrink the label with [adjustsFontSizeToFitWidth](https://developer.apple.com/documentation/uikit/uilabel/1620546-adjustsfontsizetofitwidth), or
3) use ExpandableLabel only for plain strings
image

Comment (andre1110) [reply iphone/Maps/Bookmarks/BookmarksList/BookmarksListInfoViewController.swift]: @biodranik I reconsidered :) Now I show WKWebView for both cases

Comment (biodranik) [reply iphone/Maps/Bookmarks/BookmarksList/BookmarksListInfoViewController.swift]: Can you please set a small margin and 1.25 or 1.5 line interval for plain text?

Comment (biodranik) [reply iphone/Maps/Bookmarks/BookmarksList/BookmarksListInfoViewController.swift]: Does it look better now?

Comment (andre1110) [reply iphone/Maps/Bookmarks/BookmarksList/BookmarksListInfoViewController.swift]: I think it looks better
Simulator Screenshot

Comment (andre1110) [reply iphone/Maps/Bookmarks/BookmarksList/BookmarksListInfoViewController.swift]: @biodranik

**Pull request merged**